services/datamanager/builtin/capture_test.go

```diff
@@ -42,6 +46,12 @@ func TestDataCaptureEnabled(t *testing.T) {
 		filePaths := getAllFilePaths(dir)
 		for _, path := range filePaths {
 			d, err := datacapture.SensorDataFromFilePath(path)
+			// It's possible a file was closed (and so its extension changed) in between the points where we gathered
```

Comment:
Double checking on the case where data capture is, for example, every 10 seconds.

If data is being written while the file is flushed/closed/synced, is there an immediate new target for new writes to go to?

Reply:
Yup, the data_capture_buffer will automatically handle that on the next [write](https://github.com/viamrobotics/rdk/blob/99e2f64add4c14fa40414c9903d6fb572efbfc21/services/datamanager/datacapture/data_capture_buffer.go#L57)